adapter/src/hermes/adapter/interceptor.cc

```diff
@@ -0,0 +1,95 @@
+/**
+ * Internal headers
+ */
+#include "interceptor.h"
+
+/**
+ * Define Interceptor list for adapter.
+ */
+hermes::adapter::InterceptorList list;
+
+namespace hermes::adapter {
+/**
+ * is exit by program called.
+ */
+bool exit = false;
+
+void PopulateBufferingPath() {
+  char* hermes_config = getenv(kHermesConf);
+  hermes::Config config = {};
+  const size_t kConfigMemorySize = KILOBYTES(16);
+  hermes::u8 config_memory[kConfigMemorySize];
+  if (hermes_config && strlen(hermes_config) > 0) {
+    list.hermes_paths_exclusion.push_back(hermes_config);
+    hermes::Arena config_arena = {};
+    hermes::InitArena(&config_arena, kConfigMemorySize, config_memory);
+    hermes::ParseConfig(&config_arena, hermes_config, &config);
+  } else {
+    InitDefaultConfig(&config);
+  }
+  if (!config.mount_points->empty()) {
```

Comment:
I think you just want to drop that conditional altogether. If `config.mount_points` is `["", "./nvme", "./burst_buffer"]` (which is common since the empty string represents RAM), this will fail to add the mount points to your exclusion list. You already filter out the empty string correctly on line 32. Maybe you thought `mount_points` was a `std::vector`? It's just an array of `std::string`.

Reply:
I see. ok, i removed the condition. The condition was if the array of strings was empty. Either way the for will take care of it.